x-pack/plugins/apm/public/components/app/correlations/context_popover/field_stats_popover.tsx

```diff
@@ -0,0 +1,338 @@
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+
+const HIGHLIGHTED_BUCKET_PROPS = {
+  color: 'accent',
+  textProps: {
+    color: 'accent',
+  },
+};
+export function kqlQuery(kql: string): estypes.QueryDslQueryContainer[] {
+  return !!kql ? [toElasticsearchQuery(fromKueryExpression(kql))] : [];
+}
+
+export type OnAddFilter = ({
+  fieldName,
+  fieldValue,
+  include,
+}: {
+  fieldName: string;
+  fieldValue: string | number;
+  include: boolean;
+}) => void;
+
+type FieldStatsPopoverContentProps = Omit<
+  FieldStatsProps,
+  'dataViewOrDataViewId'
+> & {
+  fieldName: string;
+  fieldValue: string | number;
+  dslQuery: object;
+  dataView: DataView;
+};
+
+export function FieldStatsPopoverContent({
+  fieldName,
+  fieldValue,
+  services,
+  field,
+  dataView,
+  dslQuery,
+  filters,
+  fromDate,
+  toDate,
+  onAddFilter,
+  overrideFieldTopValueBar,
+}: FieldStatsPopoverContentProps) {
+  const [needToFetchIndividualStat, setNeedToFetchIndividualStat] =
+    useState(false);
+
+  const onStateChange = useCallback(
+    (nextState: FieldStatsState) => {
+      const { topValues } = nextState;
+      const idxToHighlight = Array.isArray(topValues)
+        ? topValues.findIndex((value) => value.key === fieldValue)
+        : null;
+
+      setNeedToFetchIndividualStat(
+        idxToHighlight === -1 &&
+          fieldName !== undefined &&
+          fieldValue !== undefined
+      );
+    },
+    [fieldName, fieldValue]
+  );
+
+  const params = useFetchParams();
+  const { data: fieldValueStats, status } = useFetcher(
+    (callApmApi) => {
+      if (needToFetchIndividualStat) {
+        return callApmApi(
+          'GET /internal/apm/correlations/field_value_stats/transactions',
+          {
+            params: {
+              query: {
+                ...params,
+                fieldName,
+                fieldValue,
+                // Using sampler shard size to match with unified field list's default
+                samplerShardSize: '5000',
+              },
+            },
+          }
+        );
+      }
+    },
+    [params, fieldName, fieldValue, needToFetchIndividualStat]
+  );
+  const progressBarMax = fieldValueStats?.topValuesSampleSize;
+  const formatter = dataView.getFormatterForField(field);
+
+  return (
+    <>
+      <FieldStats
+        services={services}
+        field={field}
+        dataViewOrDataViewId={dataView}
+        dslQuery={dslQuery}
+        fromDate={fromDate}
+        toDate={toDate}
+        onAddFilter={onAddFilter}
+        overrideFieldTopValueBar={overrideFieldTopValueBar}
+        onStateChange={onStateChange}
+      />
+      {needToFetchIndividualStat ? (
+        <>
+          <EuiHorizontalRule margin="s" />
+          <EuiText size="xs" color="subdued">
+            <FormattedMessage
+              id="xpack.apm.correlations.fieldContextPopover.notTopTenValueMessage"
+              defaultMessage="Selected term is not in the top 10"
```

Comment:
This message should have an ending period to match "Calcuated from 5,000 messages."